src/frontend/components/check-list/check-list.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * This file is part of NER's PM Dashboard and licensed under GNU AGPLv3.
+ * See the LICENSE file in the repository root folder for details.
+ */
+
+import { ReactNode, useState } from 'react';
+import PageBlock from '../../layouts/page-block/page-block';
+import { Form, Button, Modal } from 'react-bootstrap';
+import styles from './check-list.module.css';
+
+interface CheckListProps {
+  title: string;
+  headerRight?: ReactNode;
+  list: {
```

Comment:
Could we have a more descriptive name for this prop, especially because it's asking for these custom jsx objects. I'm thinking something like listItems?